src/Illuminate/Database/Connectors/MySqlConnector.php

```diff
@@ -104,24 +104,51 @@ protected function getHostDsn(array $config)
     }
 
     /**
+     * Get the modes for the connection.
+     *
+     * @param  array  $config
+     * @return array
+     */
+    protected function getModes(array $config)
+    {
+        if (isset($config['modes'])) {
+            return (array) $config['modes'];
```

Comment:
This is in case a developer only set ONE mode as a string.
This is the only case where we get user input, so casting array cannot harm in my opinion